server-spi/src/main/java/org/keycloak/models/credential/OTPCredentialModel.java

```diff
@@ -6,22 +6,34 @@
 import org.keycloak.models.credential.dto.OTPSecretData;
 import org.keycloak.models.OTPPolicy;
 import org.keycloak.models.RealmModel;
+import org.keycloak.models.utils.Base32;
 import org.keycloak.util.JsonSerialization;
 
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 
 public class OTPCredentialModel extends CredentialModel {
 
     public static final String TYPE = "otp";
-
     public static final String TOTP = "totp";
     public static final String HOTP = "hotp";
 
+    /**
+     * The supported encodings when reading the raw secret from the storage
+     */
+    public enum SecretEncoding {
+        B32
```

Comment:
I'm bikeshedding here, but what about a more conventional `BASE32`?

Reply:
I had two issues with names in this PR. I initially started with `kEnc` for `secretEncoding` because the latter is longer. But I preferred to avoid abbreviations in this field to make it aligned with the others (e.g.: `algorithm`, etc).

I'm only using `B32` as a value to save space. No preference. Do you think it might overlap with some other encoding we might eventually support? Or we can move forward as is?

Reply:
Using `B32` will obviously just work fine, I don't foresee any conflict or anything.
I was just suggesting to use the common name of the algorithm for clarity, and not any abbreviation.
That's just 3 extra characters, so I don't think it really affect storage space.